kythe/java/com/google/devtools/kythe/analyzers/java/KytheDocTreeScanner.java

```diff
@@ -98,4 +114,36 @@ public Void visitReference(ReferenceTree tree, DCDocComment doc) {
 
     return null;
   }
+
+  @Override
+  public Void visitDeprecated(DeprecatedTree node, DCDocComment doc) {
+    if (node.getBody().isEmpty()) {
+      // deprecated tag is empty
+      return null;
+    }
+    CompilationUnitTree unit = getCurrentPath().getTreePath().getCompilationUnit();
+    DocSourcePositions positions = trees.getSourcePositions();
+    int start = (int) positions.getStartPosition(unit, doc, node.getBody().get(0));
+    int end = (int) positions.getEndPosition(unit, doc, node);
+    if (end == Position.NOPOS) {
+      // deprecated tag is empty
+      return null;
+    }
+    CharSequence source;
+    try {
+      source = unit.getSourceFile().getCharContent(/* ignoreEncodingErrors= */ true);
+    } catch (IOException e) {
+      return null;
```

Comment:
Related question here.

Reply:
I'm not saving empty strings for the other cases where the tag is empty. The case where `unit.getSourceFile().getCharContent` throws seems distinct.

It should be impossible, so I could also rethrow as a `UncheckedIOException` if you prefer.